pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/AbstractJavaRule.java

```diff
@@ -4,17 +4,21 @@
 
 package net.sourceforge.pmd.lang.java.rule;
 
+import java.util.ArrayList;
 import java.util.List;
 
 import net.sourceforge.pmd.RuleContext;
 import net.sourceforge.pmd.lang.LanguageRegistry;
 import net.sourceforge.pmd.lang.ast.Node;
 import net.sourceforge.pmd.lang.java.JavaLanguageModule;
 import net.sourceforge.pmd.lang.java.ast.*;
+import net.sourceforge.pmd.lang.java.typeresolution.TypeHelper;
 import net.sourceforge.pmd.lang.rule.AbstractRule;
 import net.sourceforge.pmd.lang.rule.ImmutableLanguage;
 
-public abstract class AbstractJavaRule extends AbstractRule implements JavaParserVisitor, ImmutableLanguage {
+public abstract class AbstractJavaRule extends AbstractRule implements JavaParserVisitor, ImmutableLanguage, Annotateable {
+
+    protected List<String> ignoredAnnotations = new ArrayList<>();
```

Comment:
As per #956, this should be a `StringMultiProperty`, so that users can configure it to add / remove annotations themselves. The default is left for child classes to configure. You can see a draft on this on https://github.com/pmd/pmd/issues/956#issuecomment-370827133